aspnetcore/blazor/fundamentals/environments.md

```diff
@@ -135,57 +152,208 @@ In the following example for IIS, the custom header (`Blazor-Environment`) is ad
+>
+> The service registrations are shown later in this section.
 
-:::moniker-end
+Add a shared class library project for the Blazor Web App:
 
-:::moniker range="< aspnetcore-5.0"
+* Visual Studio: Right-click the Blazor Web App's solution file in **Solution Explorer**. Select **Add** > **New Project** > **Class Library**. Use the name of the server project with "`.Shared`" added to the end of the name (for example: `BlazorWebAppSample.Shared`).
+* Visual Studio Code/.NET CLI: Execute `dotnet new classlib -o {PROJECT NAME}` from a command prompt in the solution's folder. The `-o|--output` option creates a folder for the class library and names the project. For the `{PROJECT NAME}` placeholder, use the name of the server project with "`.Shared`" added to the end of the name (for example: `BlazorWebAppSample.Shared`).
+
+Create a project reference for the class library in both the server and `.Client` projects:
 
-:::code language="razor" source="~/../blazor-samples/3.1/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+* Visual Studio: Right-click each project and select **Add** > **Project Reference**. Select the checkbox for the shared project and select **OK**.
+* Visual Studio Code/.NET CLI: Execute `dotnet add reference {PATH}` in a command shell from each of the server and `.Client` project folders. The `{PATH}` placeholder is the path to the shared class library project.
+
+In the shared (`.Shared`) project:
+
+* Add an interface for the service abstraction.
+* Add classes for the service implementation and host environment extensions.
+
+`BlazorHostEnvironment.cs`:
+
+```csharp
+public interface IBlazorHostEnvironment
+{
+    string EnvironmentName { get; set; }
+}
+
+public class BlazorHostEnvironment(string environment) : IBlazorHostEnvironment, 
+    IDisposable
+{
+    private bool _disposed;
+
+    public string EnvironmentName { get; set; } = environment;
+
+    public void Dispose()
+    {
+        if (_disposed)
+        {
+            return;
+        }
+
+        _disposed = true;
+    }
+}
+
+public static class BlazorHostEnvironmentExtensions
+{
+    public static bool IsDevelopment(this IBlazorHostEnvironment hostEnvironment)
+    {
+        return hostEnvironment.IsEnvironment("Development");
+    }
+
+    public static bool IsStaging(this IBlazorHostEnvironment hostEnvironment)
+    {
+        return hostEnvironment.IsEnvironment("Staging");
+    }
+
+    public static bool IsProduction(this IBlazorHostEnvironment hostEnvironment)
+    {
+        return hostEnvironment.IsEnvironment("Production");
+    }
+
+    public static bool IsEnvironment(
+        this IBlazorHostEnvironment hostEnvironment,
+        string environmentName)
+    {
+        return string.Equals(
+            hostEnvironment.EnvironmentName,
+            environmentName,
+            StringComparison.OrdinalIgnoreCase);
+    }
+}
```

Comment:
Well ... it's kind'a nice to get the methods. I wrote this so that devs can just use it just like this ... save them some time. It's fast to delete the bits they don't want. It's slower (and a pain) for them to have to create these.